daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/Runner.scala

```diff
@@ -356,7 +356,7 @@ object Runner {
       timeMode: ScriptTimeMode,
       traceLog: TraceLog = Speedy.Machine.newTraceLog,
       warningLog: WarningLog = Speedy.Machine.newWarningLog,
-      canceled: () => Boolean = () => false,
+      canceled: () => Option[Boolean] = () => None,
```

Comment:
I'm a bit worried about boolean blindness here, perhaps this could instead be

```suggestion
      canceled: () => Option[RuntimeException] = () => None,
```

that way, we can use plain `Canceled` for manually cancelled runs (`Canceled(false)`) and add `TimedOut` for `Canceled(true)`

Reply:
Fixing now

Reply:
Fixed